transformer_engine/pytorch/module.py

```diff
@@ -541,7 +541,9 @@ def prepare_forward(
 
         if self.fp8 and self.training and self.fp8_meta["recipe"].reduce_amax:
             set_fp8_context_id(self.fp8_meta["autocast_id_fwd"])
-            reduce_func = partial(global_amax_reduction, self.fp8_meta, forward=True)
+            reduce_func = partial(
+                global_amax_reduction, self.fp8_meta, self.tp_group, forward=True
```

Comment:
If the user hasn't initialized tensor parallelism, won't this pass `None` into `global_amax_reduction`? This would cause global communication in the steps where we expect no communication. `torch.distributed` treats `None` as the world communicator and I don't see a nice way to get the self communicator, so I think we should pass `self.tp_group_initialized` into `global_amax_reduction`.

Reply:
Makes sense. Do you think that is preferred over skipping reduction when `tp_group=None`?

Reply:
Pushed an update such that the reduction is skipped when TP-group is not initialized.